ruby/ext/google/protobuf_c/message.c

```diff
@@ -192,7 +192,7 @@ static int extract_method_call(VALUE method_name, MessageHeader* self,
     // Find the field name
     char wrapper_field_name[name_len - 8];
     strncpy(wrapper_field_name, name, name_len - 9);
-    wrapper_field_name[name_len - 7] = '\0';
+    wrapper_field_name[name_len - 9] = '\0';
```

Comment:
https://github.com/protocolbuffers/protobuf/blob/8041dd4034dc837f14c92846acd669e4df01cae2/ruby/ext/google/protobuf_c/message.c#L295

Previously, it should be interpreted as METHOD_UNKNOWN. Then, the value will be set to the parent ruby object.
I guess, if you actually encode the message then, the value you set won't appear in the encoding result.